amethyst_rendy/src/camera.rs

```diff
@@ -395,3 +423,258 @@ impl<'a> PrefabData<'a> for ActiveCameraPrefab {
+        let camera_transform : Transform = Transform::new(
+            Translation3::new(0.0, 0.0, -3.0), 
+            UnitQuaternion::from_axis_angle(&Vector3::y_axis(), 0.0),
+            // UnitQuaternion::from_axis_angle(&Vector3::y_axis(), std::f32::consts::PI),
+            [1.0, 1.0, 1.0].into());
+
+        let simple_points : [Point3<f32>; 3] = [
+            Point3::new(1.0, 0.0, 0.0),
+            Point3::new(0.0, 1.0, 0.0),
+            Point3::new(0.0, 0.0, 1.0)
+        ];
+
+        let simple_points_clipped : [Point3<f32>; 3] = [
+            Point3::new(-20.0, 0.0, 0.0),
+            Point3::new(0.0, -20.0, 0.0),
+            Point3::new(0.0, 0.0, -10.0)
+        ];
+        (camera_transform, simple_points, simple_points_clipped)
+    }
+
+    fn gatherer_calc_view_matrix(transform: Transform) -> Matrix4<f32> {
+        // let mat : Matrix4<f32> = convert(*transform.global_matrix());
+        // mat.try_inverse().expect("Unable to get inverse of camera transform")
+        convert(transform.view_matrix())
+    }
+
+    #[test]
+    fn camera_matrix() {
+        let (camera_transform, _, _) = setup();
+        let iso = Isometry3::face_towards(
+            &Point3::new(0.0, 0.0, -3.0), 
+            &Point3::new(0.0, 0.0, 0.0), 
+            &Vector3::y_axis()
+        );
+        let our_iso : Matrix4<f32> = convert(camera_transform.isometry().to_homogeneous());
+        // Check camera isometry
+        assert_ulps_eq!(our_iso, iso.to_homogeneous());
+
+        let view_matrix = Isometry3::look_at_lh(
+            &Point3::new(0.0, 0.0, -3.0), 
+            &Point3::new(0.0, 0.0, 0.0), 
+            &Vector3::y_axis()
+        );
+        // Check view matrix.
+        // The view matrix is used to transfrom a point from world space to eye space.
+        // Changes the base of a vector from world origin to your eye.
+        let our_view : Matrix4<f32> = convert(camera_transform.view_matrix());
+        assert_ulps_eq!(our_view, view_matrix.to_homogeneous(), max_ulps = 10);
+
+        let our_inverse = gatherer_calc_view_matrix(camera_transform);
+        assert_ulps_eq!(our_inverse, view_matrix.to_homogeneous());
+
+        assert_ulps_eq!(our_inverse, our_view);
+    }
+
+    #[test]
+    fn standard_2d() {
+        unimplemented!();
+    }
+
+    #[test]
+    fn standard_3d() {
+        unimplemented!();
+    }
+
+
+    #[test]
+    fn perspective_orientation() {
+        let (camera_transform, simple_points, simple_points_clipped) = setup();
+
+        let proj = Projection::perspective(1280.0/720.0, std::f32::consts::FRAC_PI_3, 0.1, 100.0);
+        let view = gatherer_calc_view_matrix(camera_transform);
+
+        let mvp = proj.as_matrix() * view;
+
+        let x_axis = mvp * simple_points[0].to_homogeneous();
+        let y_axis = mvp * simple_points[1].to_homogeneous();
+        let z_axis = mvp * simple_points[2].to_homogeneous();
+
+        dbg!(x_axis);
+        assert!(x_axis[0] > 0.0);
```

Comment:
Not a fan of `dbg!`. This shouldn't ever be present in commited code. You can load up some crate to do something like `assert_gt!` instead.